heapsort/heapsort.py

```diff
@@ -0,0 +1,25 @@
+def s(l):
+    #size
+    n = len(l)
+    print("n:",n)
+    print("range:", range(n,-1,-1))
+    #build heap
+    for i in range(n,-1,-1):
```

Comment:
I did not carefully review your heap implementation here but it does not seem correct to me at first glance because your l_idx and r_idx indices go very out of range. There should also be more than one loop in your heap function. If you are using the "sift down" approach, the first loop is going through all nodes in reverse order as you are doing, and there should be a second loop that iterates through all children of the the node that's being processed.